iam.tf

```diff
@@ -66,3 +66,44 @@ resource "aws_iam_policy_attachment" "attach-ecs" {
   roles      = ["${aws_iam_role.ecs-role.name}"]
   policy_arn = "${aws_iam_policy.ecs-policy.arn}"
 }
+
+# IAM Resources for Consul and Registrator Agents
+
+data "aws_iam_policy_document" "consul_task_policy" {
+  statement {
+    actions = [
+      "ec2:Describe*",
+      "autoscaling:Describe*",
+      "ec2:DescribeAddresses",
+      "ec2:DescribeInstances",
+      "ec2:DescribeTags",
+    ]
+
+    resources = ["*"]
+  }
+}
+
+data "aws_iam_policy_document" "assume_role_consul_task" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "consul_task" {
+  count              = "${var.enable_agent ? 1 : 0}"
+  name               = "tf-${data.aws_vpc.vpc.tags["Name"]}-consul-agentTaskRole"
```

Comment:
😿   🎺   `* module.infra-svc-pub.aws_iam_role.consul_task: "name_prefix" cannot be longer than 32 characters, name is limited to 64`